2_process/src/raster_to_catchment_polygons.R

```diff
@@ -0,0 +1,89 @@
+rbind.fill.list <- function(x) {
+  nam <- sapply(x, names)
+  unam <- unique(unlist(nam))
+  len <- sapply(x, length)
+  out <- vector("list", length(len))
+  for (i in seq_along(len)) {
+    out[[i]] <- unname(x[[i]])[match(unam, nam[[i]])]
+  }
+  setNames(as.data.frame(do.call(rbind, out), stringsAsFactors=FALSE), unam)
+}
+
+raster_to_catchment_polygons <- function(polygon_sf, raster,
+                                         categorical_raster = NULL,
+                                         raster_summary_fun = NULL,
+                                         new_cols_prefix = NULL){
+
+  #' @description Function extracts raster pixel per catchment polygon and calculated proportion of raster class per total
+  #'
+  #' @param polygon_sf sf object of polygon(s) in aoi
+  #' @param raster path to '.tif' file or raster object of SpatRaster object  
+  #' @param categorical_raster logical. If categorical raster, TRUE
+  #' @param raster_summary_fun for continuous raster, function to summarize the data by geometry. kept NULL if categorical_raster = TRUE
+  #' @param categorical_raster_prefix for categorical raster, prefix added to colnames to label raster classes/values
```

Comment:
Should `@param categorical_raster_prefix` be replaced with `new_cols_prefix` here, or is that a separate parameter?

Reply:
yes this was old documentation - pre rd salt. thanks!